test/reClammPool.test.ts

```diff
@@ -199,18 +217,65 @@ describe('ReClammPool', function () {
       }
     );
 
-    // make the next swap
-    await router.connect(bob).swapSingleTokenExactIn(
-      pool, // pool address
-      tokenB, // token we're swapping from
-      tokenA, // token we're swapping to
-      exactAmountIn, // exact amount we're swapping
-      minAmountOut, // minimum amount we want to receive
-      deadline, // deadline for the swap
-      wethIsEth, // whether we're dealing with ETH
-      '0x' // no additional data
+    expect(expectedFinalVirtualBalances[tokenAIdx]).to.be.greaterThan(virtualBalancesAfterSwap[tokenAIdx]);
+    expect(expectedFinalVirtualBalances[tokenBIdx]).to.be.lessThan(virtualBalancesAfterSwap[tokenBIdx]);
+
+    // Swap in the other direction
+    await router
+      .connect(bob)
+      .swapSingleTokenExactOut(pool, tokenB, tokenA, INITIAL_BALANCE_A, MAX_UINT256, deadline, wethIsEth, '0x');
+
+    // check if the virtual balances are close from expected
+    const actualFinalVirtualBalances = await pool.getCurrentVirtualBalances();
+
+    expect(actualFinalVirtualBalances.length).to.be.equal(2);
+    expectEqualWithError(actualFinalVirtualBalances[0], expectedFinalVirtualBalances[0], virtualBalancesError);
+    expectEqualWithError(actualFinalVirtualBalances[1], expectedFinalVirtualBalances[1], virtualBalancesError);
+  });
+
+  it('should move virtual balances correctly (out of range < center)', async () => {
+    // Very big swap, putting the pool right at the edge.
+    const exactAmountOut = INITIAL_BALANCE_A - MIN_POOL_BALANCE - 1n;
+    const maxAmountIn = MAX_UINT256;
+    const deadline = MAX_UINT256;
+    const wethIsEth = false;
+
+    await router
+      .connect(bob)
+      .swapSingleTokenExactOut(pool, tokenB, tokenA, exactAmountOut, maxAmountIn, deadline, wethIsEth, '0x');
+
+    const [, , , poolBalancesAfterSwap] = await vault.getPoolTokenInfo(pool);
+    const virtualBalancesAfterSwap = await pool.getCurrentVirtualBalances();
+    const lastTimestamp = (await ethers.provider.getBlock('latest')).timestamp;
+
+    // Pass 1 hour
+    await ethers.provider.send('evm_increaseTime', [60 * 60]);
+    await ethers.provider.send('evm_mine');
```

Comment:
```suggestion
    const lastTimestamp = await currentTimestamp();

    await advanceTime(HOUR);

    const expectedTimestamp = lastTimestamp + BigInt(HOUR) + 1n;
```
We have existing helpers for this. Can just use `expectedTimestamp` in the virtual balances call below.

Reply:
Nice to know!